src/app/epics/user-login.epic.spec.ts

```diff
@@ -0,0 +1,176 @@
+import { UserLoginEpic } from './user-login.epic';
+import { ofType, ActionsObservable } from 'redux-observable';
+import { Observable } from 'rxjs/Observable';
+import 'rxjs/add/observable/of';
+import 'rxjs/add/observable/throw';
+
+import 'rxjs/add/observable/empty';
+import 'rxjs/add/operator/switchMap';
+import 'rxjs/add/operator/map';
+import 'rxjs/add/operator/catch';
+
+import { AnyAction } from 'redux';
+import { IUser } from '../models/user.model';
+import { UserActions } from '../actions/user.actions';
+
+let mockUser: IUser = { name: 'MOCK_user' };
+
+class MockLoginService {
+    login = (username, pass) => {};
+
+    logout = () => {};
+};
+// TODO: Don't feel confident in the testing of Observables.
+// TODO: Needs to be updated since Login Service has changed its error handle.
+//      And because there is now a Router dependency
+class MockUserActions {
+    loginSuccess = (user, date) => ({ type: 'MOCK_LOGIN_SUCCESS', user, date });
+    loginFailed = (err) => ({ type: 'MOCK_LOGIN_ERROR', err });
+};
+
+class MockRouter {
+    router = {
+        navigate: (commands, extras) => {},
+    };
+}
+
+
+describe('UserLoginEpic', () => {
+    let mockUserActions: MockUserActions;
+    let mockLoginService: MockLoginService;
+    let mockRouter: MockRouter;
+    let userLoginEpic: UserLoginEpic;
+    let action$: any;
+
+    
+
+    describe('login', () => {
+        beforeEach(() => {
+            mockLoginService = new MockLoginService();
+            mockUserActions = new MockUserActions();
+            mockRouter = new MockRouter();
+            userLoginEpic = new UserLoginEpic(
+                mockLoginService as any, 
+                mockUserActions as any,
+                mockRouter as any
+            );
+        });
+
+        it('should initiate login and succeed', (done: DoneFn) => {
+            const user: string = 'USER';
+            const pass: string = 'PASSWORD';
+            const date: number = Date.now();
+
+            spyOn(mockLoginService, 'login')
+                .and
+                .returnValue( Observable.of(mockUser) );
+            spyOn(mockUserActions, 'loginSuccess')
+                .and
+                .returnValue( mockUserActions.loginSuccess(user, date) );
```

Comment:
Based on feedback below, since we'd not longer be asserting that these specific action creator methods would be invoked. Since these values are already returning mock actions anyway, spying on them just to return their invocations doesn't make as much sense.

If you had a stubbed out object, you could do something like... `spyOn(mockUserActions, 'loginSuccess').and.returnValue(mockLoginSuccessAction)`

In this case your mock already returns a mock value, so no need.